### ISSUE TITLE
Data race for `LeastBytes.Balance()`

### ISSUE THREAD

**Issue by hansbogert** [segmentio/kafka-go#647]
**Describe the bug**
Data race when using `WriteMessages(..)` concurrently icw `LeastBytes`

**Kafka Version**
What version(s) of Kafka are you testing against?
```
docker-compose config | grep kafka | grep image
    image: bitnami/kafka:2.7.0
```

**To Reproduce**
have multiple goroutines use `WriteMessages(..)` icw `LeastBytes`

**Expected behavior**
`WriteMessages(..)` should be allowed to be called concurrently as per the documentation at  [1](https://github.com/segmentio/kafka-go/blob/fbe3d96cd21174e8d8ef6ce43ab10f2bb21e722a/writer.go#L76). By transitivity, this should also be the case for `LeastBytes.Balance()`, as it is called in `WriteMessages()` 

**Additional context**
```
WARNING: DATA RACE
Read at 0x00c00100d668 by goroutine 476:
  github.com/segmentio/kafka-go.(*LeastBytes).Balance()
      /go/pkg/mod/github.com/segmentio/kafka-go@v0.4.10/balancer.go:80 +0xba
  github.com/segmentio/kafka-go.(*Writer).WriteMessages()
      /go/pkg/mod/github.com/segmentio/kafka-go@v0.4.10/writer.go:605 +0x7b8
  github.com/looplab/eventhorizon/eventbus/kafka.(*EventBus).HandleEvent()
     [redacted]

Previous write at 0x00c00100d668 by goroutine 479:
  github.com/segmentio/kafka-go.(*LeastBytes).Balance()
      /go/pkg/mod/github.com/segmentio/kafka-go@v0.4.10/balancer.go:91 +0x215
  github.com/segmentio/kafka-go.(*Writer).WriteMessages()
      /go/pkg/mod/github.com/segmentio/kafka-go@v0.4.10/writer.go:605 +0x7b8
  github.com/looplab/eventhorizon/eventbus/kafka.(*EventBus).HandleEvent()
     [redacted]

```

The writing on the wall seems to be that `mutex` in `LeastBytes` is not used at all.

**Comment by hansbogert** [comment]
Proposed solution #648

**Comment by stevevls** [comment]
Closed by #648
